Troubleshooting section added for those that lack rbenv in their path

Bastes opened the pull request:
Following [issue #9](https://github.com/oh-my-fish/plugin-rbenv/issues/9), I thought people encountering the same problem would benefit from a troubleshooting section explaining the issue and how to fix it.

Bastes committed:
Better way to add $RBENV_ROOT to the path before testing for the rbenv command

bpinto commented on README.md:
```diff
@@ -26,6 +26,26 @@ If you have a custom rbenv root, you can add the following to your `~/.config/fi
+...
+
+# assuming your rbenv command is located under ~/.rbenv/bin
+set -g PATH $HOME/.rbenv/bin $PATH
```
No need for -g

Bastes replied:
Thanks, fixing it right away.

bpinto commented:
Shouldn't we do this automatically inside this if block?

```
  if not set -q RBENV_ROOT
    set -x RBENV_ROOT "$HOME/.rbenv"
  end
```

Bastes commented:
@bpinto I'm not sure what you meant there ^^° but if you mean replace the "available rbenv" check and instead put something that looks up the $RBENV_ROOT path for the rbenv executable, I'm all for it. I'll try and give you a new PR when I'm done.

bpinto commented:
Ping me once you have a PR, I'll be on a computer in the next few days so I'll become more helpful!! Replying on a mobile is a terrible experience.

Bastes commented:
@bpinto terrible indeed ^^ stay tuned for the PR

Bastes commented:
@bpinto There ; I added what we discussed (unless I've missed the point ^^°) and removed the now redundant troubleshooting section in the README ; it fixes the problem on my config at least (provided of course I added the $RBENV_ROOT variable in my fish.config) but it feels a bit patchy. Feel free to comment ^^

alessandro-fazzi commented:
In order to avoid the arrow anti-pattern I'd suggest to do an easier to read (really not tested!)

``` fish
if set -q RBENV_ROOT and not contains "$RBENV_ROOT/bin" $PATH
  set PATH $RBENV_ROOT/bin $PATH
end

if not available rbenv
  echo "Please install 'rbenv' first, or set \$RBENV_ROOT!"; return 1
end
```

also if less sophisticated. What do you think?

Thanks for the PR.

Cheers :)

Bastes commented:
@pioneerskies That might just work ^^ I'll test and send a more sophisticated PR when it passes. Thanks.

bpinto commented:
That's indeed what I was thinking!

You could also use autoload function as it prevents duplicated paths too. Not sure what is better...

Bastes commented:
@pioneerskies @bpinto Here goes, tested on my config ^^

@bpinto I'm not sure what to make of the autoload command, I found it in OMF's internals but I'm not quite sure what it does and how to use it (I tried to read the source but I guess I'm a bit too green to shell scripting to really get it ^^°).

bpinto commented on rbenv.fish:
```diff
@@ -2,8 +2,12 @@
 #   rbenv plugin for oh my fish
 
 function init --on-event init_rbenv
+  if not available rbenv; and set -q RBENV_ROOT; and not contains "$RBENV_ROOT/bin" $PATH
```
Unnecessary check for rbenv command?

Bastes replied:
It feels a little redudant, and it works without, so I'm not adamant about keeping it. It just felt a little redundant to add $RBENV_ROOT to the $PATH when the rbenv command is already available.

bpinto commented:
Could you squash your commits?

bpinto commented:
I could walk you through the code next week if you are interested on understand that function.

Bastes commented:
@bpinto Squashing right away ^^ and I'd certainly be interested to know more about autoload anytime.

Bastes commented:
@bpinto There it is, all squashy as it should.

bpinto commented:
Super! Thanks for the great work done here!

Bastes commented:
Thank you all for the guidance ^^
